arch/arm/core/thread.c

```diff
@@ -93,5 +94,32 @@ void _new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	 * irrelevant.
 	 */
 
+#ifdef CONFIG_THREAD_MONITOR
+	/*
+	 * In debug mode thread->entry give direct access to the thread entry
+	 * and the corresponding parameters.
+	 */
+	thread->entry = (struct __thread_entry *)(pInitCtx);
 	thread_monitor_init(thread);
+#endif
+}
+
+#ifdef CONFIG_ARM_USERSPACE
+
+FUNC_NORETURN void _arch_user_mode_enter(k_thread_entry_t user_entry,
+                                       void *p1, void *p2, void *p3)
+{
+	_arm_userspace_enter(user_entry, p1, p2, p3,
+			     (u32_t)_current->stack_obj,
+			     _current->stack_info.size);
+	CODE_UNREACHABLE;
```

Comment:
It would be better to add k_thread_abort() here like what _thread_entry() does to
handle the case that thread returns from user_entry() unexpectedly.

Reply:
NACK
that isn't how this works, user_entry will never return here

Reply:
Yeah, I got it after reviewing the x86 implementation.